File: frontend/src/microservices/flight/objects/FlightSearchPlannerRequest.ts
// 本文件定义 flight 模块的 `FlightSearchPlannerRequest`，作为 planner 请求参数并提供 JSON 编解码。

export type FlightSearchPlannerRequest = {
  departureAirport?: string
  arrivalAirport?: string
  date?: string
}

export const flightSearchPlannerRequestFromJson = (json: string): FlightSearchPlannerRequest =>
  JSON.parse(json) as FlightSearchPlannerRequest

export const flightSearchPlannerRequestToJson = (value: FlightSearchPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/FlightSearchPlannerRequest.scala
// FlightSearchPlannerRequest defines the flight search planner request model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightSearchPlannerRequest(
    departureAirport: Option[String],
    arrivalAirport: Option[String],
    date: Option[String]
)

object FlightSearchPlannerRequest:
  given sourceEncoder: Encoder[FlightSearchPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[FlightSearchPlannerRequest] = deriveDecoder

